instrumentation/opentelemetry-instrumentation-asyncpg/tests/test_asyncpg_wrapper.py

```diff
@@ -34,3 +39,52 @@ def test_duplicated_uninstrumentation(self):
             self.assertFalse(
                 hasattr(method, "_opentelemetry_ext_asyncpg_applied")
             )
+
+    def test_cursor_instrumentation(self):
+        def assert_wrapped(assert_fnc):
+            for cls, methods in [
+                (cursor.Cursor, ("forward", "fetch", "fetchrow")), 
+                (cursor.CursorIterator, ("__anext__", ))
+            ]:
+                for method_name in methods:
+                    method = getattr(cls, method_name, None)
+                    assert_fnc(isinstance(method, ObjectProxy), f"{method} isinstance {type(method)}")
+
+
+        assert_wrapped(self.assertFalse)
+        AsyncPGInstrumentor().instrument()
+        assert_wrapped(self.assertTrue)
+        AsyncPGInstrumentor().uninstrument()
+        assert_wrapped(self.assertFalse)
+
+    def test_cursor_span_creation(self):
+        """ Test the cursor wrapper if it creates spans correctly.
+        """
+        async def mock_fn(*args, **kwargs):
+            pass
+
+        async def mock_fn_stop(*args, **kwargs):
+            raise StopAsyncIteration()
+
+        cursor_mock = mock.Mock()
+        cursor_mock._query = "SELECT * FROM test"
+
+        apg = AsyncPGInstrumentor()
+        apg.instrument(tracer_provider=self.tracer_provider)
+
+        # Call the wrapper function directly. They only way to be able to do this on the real classes is to mock all of the
```

Comment:
If you update this to use the wrapped function, as most of the other tests in this repository I guess, I'll approve and eventually move things forward

Reply:
I now test the functions directly. This requires to mock out all connection related handling in asyncpy.cursor